packages/experimental/smart-session/src/core/utils/TypeUtils.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * A wallet is the signer for these permissions
+ * `data` is not necessary for this signer type as the wallet is both the signer and grantor of these permissions
+ */
+export type WalletSigner = {
+  type: 'wallet'
+  data: Record<string, unknown>
+}
+
+// The types of keys that are supported for the following `key` and `keys` signer types.
+export type KeyType = 'secp256r1' | 'secp256k1' | 'ed25519' | 'schnorr'
+
+/*
+ * A signer representing a single key.
+ * "Key" types are explicitly secp256r1 (p256) or secp256k1, and the public keys are hex-encoded.
+ */
+export type KeySigner = {
+  type: 'key'
+  data: {
+    type: KeyType
+    publicKey: `0x${string}`
+  }
+}
+
+/*
+ * A signer representing a multisig signer.
+ * Each element of `publicKeys` are all explicitly the same `KeyType`, and the public keys are hex-encoded.
+ */
+export type MultiKeySigner = {
+  type: 'keys'
+  data: {
+    keys: {
+      type: KeyType
+      publicKey: `0x${string}`
+    }[]
+  }
+}
+
+// An account that can be granted with permissions as in ERC-7710.
+export type AccountSigner = {
+  type: 'account'
+  data: {
+    address: `0x${string}`
+  }
+}
+
+export type Signer = WalletSigner | KeySigner | MultiKeySigner | AccountSigner
```

Comment:
We don't support all of these, we need only `MultiKeySigner` for now

Reply:
We will need KeySigner and MultiKeySigner both. 
We are internally converting the type to MultiKeySigner.

Reply:
changed to just `MultiKeySigner` on Appkitsdk.